CMakeLists.txt.in

```diff
@@ -0,0 +1,298 @@
+  RESULT build_concurrency_factor
+  QUERY NUMBER_OF_LOGICAL_CORES)
+set(make_command make -j ${build_concurrency_factor})
+
+set(autotools_flags
+  CFLAGS=-I@SMF_DEPS_INSTALL_DIR@/include
+  CXXFLAGS=-I@SMF_DEPS_INSTALL_DIR@/include
+  LDFLAGS=-L@SMF_DEPS_INSTALL_DIR@/lib)
+
+set(PKG_CONFIG_PATH PKG_CONFIG_PATH=@SMF_DEPS_INSTALL_DIR@/lib/pkgconfig)
+
+set(info_dir --infodir=@SMF_DEPS_INSTALL_DIR@/share/info)
+
+include(ExternalProject)
+
+ExternalProject_Add(boost_ext
+  # The 1.67.0 release has a bug in Boost Lockfree around a missing header.
+  URL https://dl.bintray.com/boostorg/release/1.64.0/source/boost_1_64_0.tar.gz
+  URL_MD5 319c6ffbbeccc366f14bb68767a6db79
+  PATCH_COMMAND
+    ./bootstrap.sh
+    --prefix=@SMF_DEPS_INSTALL_DIR@
+    --with-libraries=atomic,chrono,date_time,filesystem,program_options,system,test,thread
+  CONFIGURE_COMMAND ""
+  BUILD_COMMAND ""
+  INSTALL_COMMAND
+    ${CMAKE_COMMAND} -E chdir <SOURCE_DIR>
+    ./b2
+    -j ${build_concurrency_factor}
+    --layout=system
+    --build-dir=<BINARY_DIR>
+    install
+    variant=release
+    link=static
+    threading=multi
+    hardcode-dll-paths=true
+    dll-path=@SMF_DEPS_INSTALL_DIR@/lib)
+
+ExternalProject_Add(fmt_ext
+  URL https://github.com/fmtlib/fmt/archive/9e554999ce02cf86fcdfe74fe740c4fe3f5a56d5.tar.gz
+  URL_MD5 e2869a5032a6cc35b4d14008b6125ed9
+  INSTALL_DIR    @SMF_DEPS_INSTALL_DIR@
+  CMAKE_ARGS
+    -DCMAKE_BUILD_TYPE=Release
+    -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+    -DFMT_INSTALL=ON
+    -DFMT_DOC=OFF
+    -DFMT_TEST=OFF
+  BUILD_COMMAND ${make_command})
+
+ExternalProject_Add(protobuf_ext
+  URL https://github.com/protocolbuffers/protobuf/releases/download/v3.3.0/protobuf-cpp-3.3.0.tar.gz
+  URL_MD5 73c28d3044e89782bdc8d9fdcfbb5792
+  INSTALL_DIR @SMF_DEPS_INSTALL_DIR@
+  CONFIGURE_COMMAND <SOURCE_DIR>/configure --prefix=<INSTALL_DIR> --srcdir=<SOURCE_DIR>
+  BUILD_COMMAND ${make_command})
+
+# FIXME: this setup seems to cause dpdk to rebuild each time
+set(dpdk_quadruple ${CMAKE_SYSTEM_PROCESSOR}-native-linuxapp-gcc)
+
+set (dpdk_args
+  EXTRA_CFLAGS=-Wno-error
+  O=<BINARY_DIR>
+  DESTDIR=@SMF_DEPS_INSTALL_DIR@
+  T=${dpdk_quadruple})
+
+ExternalProject_Add(dpdk_ext
+  URL https://github.com/scylladb/dpdk/archive/a1774652fbbb1fe7c0ff392d5e66de60a0154df6.tar.gz
+  URL_MD5 2af337f9ed29955362845808d6cdaa3e
+  CONFIGURE_COMMAND
+    COMMAND
+      ${CMAKE_COMMAND} -E chdir <SOURCE_DIR>
+      make ${dpdk_args} config
+    COMMAND
+      ${CMAKE_COMMAND}
+      -DSeastar_DPDK_CONFIG_FILE_IN=<BINARY_DIR>/.config
+      -DSeastar_DPDK_CONFIG_FILE_CHANGES=@CMAKE_CURRENT_SOURCE_DIR@/dpdk_config
+      -DSeastar_DPDK_CONFIG_FILE_OUT=<BINARY_DIR>/${dpdk_quadruple}/.config
+      -P @CMAKE_CURRENT_SOURCE_DIR@/cmake/dpdk_configure.cmake
+  BUILD_COMMAND ""
+  INSTALL_COMMAND
+   ${CMAKE_COMMAND} -E chdir <SOURCE_DIR>
+   ${make_command} ${dpdk_args} install)
+
+ExternalProject_Add(c-ares_ext
+  URL https://c-ares.haxx.se/download/c-ares-1.13.0.tar.gz
```

Comment:
U know, i was looking at c-ares and it is also all in github and w/ a proper cmake module, so i think we should use that instead of the configure one. Thoughts? 

It looks like a legit cmake project too with proper configs.

Reply:
yeh, much prefer to avoid autotools when possible. to make all this tractable i was avoiding thinking about versions and pinning them to the versions used in seastar. IIRC c-ares v1.13.0 had a cmake issue. I'll look at it again, but in general what's your take on upgrading some of the dependencies?

Reply:
ah, grabbing the 1.13.0 tarball of github has cmake support. the 1.13.0 release published from haxx.se doesn't have cmake for some reason. question about upgrade still stands.

Reply:
For the ones that we are providing and downloading and building manually, I'm Ok taking a later version (our minimum version is really dictated by seastar)

Reply:
fixed